### PR TITLE
fix(regexp): scope nullable-quantifier rewrite to nullable alternation branches

### LINKED ISSUE #370: Nullable-lazy-marker rewrite drops matched text on alternation branches (e.g. /(a*|dc??)*/)
## Summary

Discovered while dual-evaluator-triaging reviewer feedback on PR #368 (unrelated to that PR's brace-literal scope — this reproduces with no `{`/`}` in the pattern at all, so it's a separate, pre-existing bug in the same `fix_nullable_quantifiers` nullable-rewrite pass).

## Repro

```
$ node -e 'let m = /(a*|dc??)*/.exec("dc"); console.log(JSON.stringify(m[0]));'
"d"

$ jsse -e 'let m = /(a*|dc??)*/.exec("dc"); console.log(JSON.stringify(m[0]));'
""
```

Node (spec-correct) matches `"d"`; jsse matches the empty string, i.e. the outer `(...)*` group is matching zero times instead of once.

## Likely area

`src/interpreter/builtins/regexp.rs` — `fix_nullable_quantifiers`/`nq_is_nullable`/`nq_mark_lazy`. `nq_is_nullable` treats `|` specially by returning `true` as soon as it sees the alternation operator ("if ANY alternative is nullable, the whole alternation is nullable" — see the comment at the `'|'` match arm), which is correct for *nullability* but `nq_mark_lazy`'s corresponding lazy-marker-stripping logic for the repeated group appears to not correctly account for alternation branches — it seems to strip the lazy `??` marker on the `dc??` branch even though that specific branch is not itself nullable (only the `a*` branch is), causing the outer `*` to treat the whole group as always matching empty.

## Discovered via

Automated evaluation during `/pm-autofix-pr` processing of PR #368 (jsse#359's brace-literal fix). Not filed as part of that PR since it doesn't involve braces and is out of scope for it.

### PR DESCRIPTION
**Pull request by pmatos**
## Summary

- `/(a*|dc??)*/.exec("dc")` matched `""` instead of `"d"` (Node). Per spec §22.2.2.6.1 step 2.b, an iteration of a repeated group that matches empty must be discarded — but `regex`/`fancy-regex` commit to the loop's first zero-width success instead of backtracking, so a nullable alternation branch (`a*`) could pre-empt a sibling branch able to consume input (`dc??`) at the same position.
- The existing `fix_nullable_quantifiers` rewrite (added for the single-alternative case, e.g. `/(a?b??)*/`) also stripped lazy markers off *every* alternation branch indiscriminately, not just the ones that actually make the group nullable — a real bug in its own right, independently confirmed against Node.
- Fix: scope the lazy-strip per branch (only nullable branches get touched), and for a branch that reduces to a single min-0-quantified atom (`a*`, `a?`, `a{0,n}`), forbid the zero-occurrence case outright (`a*`→`a+`, `a?`→`a`, `a{0,n}`→`a{1,n}`) — exactly the iteration spec 2.b already discards, so every non-empty match the branch could produce is unaffected, and no parentheses move (capture-group numbering is untouched).
- An **alternative fix (reordering nullable branches to the end) was tried and rejected**: it's unsound — `/(a*|ab)*/.exec("ab")` is `"a"` on Node (leftmost branch wins even though it's nullable), but reordering to `/(ab|a*)*/` changes that to `"ab"`. Verified via a throwaway probe against both `regex` and `fancy-regex` directly before committing to the bump-in-place approach instead.
- Single-alternative bodies are left exactly as before (still gated on lazy-marker presence) — they don't have this bug (no sibling branch to lose priority to), so touching them was unnecessary risk.
- A few related nullable-alternation shapes (bare empty branch `(|a)*`, jointly-optional compound branch `a?b?`, a nullable atom with no quantifier suffix of its own, and the `{0}`/`{0,0}` exact-zero edge case) are **not** fixed here — confirmed via a Node-diff matrix to be pre-existing, not regressions — and tracked in #373.

Fixes #370

## Test plan

- [x] `cargo test --release` — 341 unit tests + test262 smoke oracle, all passing
- [x] `./scripts/lint.sh` — rustfmt + clippy clean
- [x] `uv run python scripts/run-test262.py -j 32` — 100.00% (99,568/99,568), 0 regressions, 323 new passes
- [x] `uv run python scripts/run-test262.py test262/test/staging/ -j 32` — 98.54% (2,706/2,746), 0 regressions (matches known baseline)
- [x] `uv run python scripts/run-custom-tests.py` — 9/9
- [x] `uv run python scripts/run-test262.py test262-extra/` — new `RegExp-nullable-alternation-quantifier.js` passes; 2 pre-existing unrelated failures (documented `noStrict`-flag gaps)
- [x] Manual repro confirms the bug is fixed (`/(a*|dc??)*/.exec("dc")[0] === "d"`, matching Node)
- [x] New regression test added (`test262-extra/RegExp-nullable-alternation-quantifier.js`), covering the reported repro, a no-laziness variant sharing the same root cause, bounded `{0,n}`/`{0,}` quantifiers, the rejected-reordering counterexample, and capture-group numbering

### CODE REVIEW

File: src/interpreter/builtins/regexp.rs [hunk 1]
```diff
@@ -5898,23 +5915,202 @@ fn fix_nullable_quantifiers(source: &str) -> String {
...
+            continue;
+        }
+        for (bstart, bend) in branches {
+            if bstart >= bend {
+                continue; // bare empty alternative — not handled here (jsse#370 follow-up)
+            }
+            if !nq_is_nullable(&chars, bstart, bend, &close_of) {
+                continue;
+            }
+            if !nq_bump_bare_atom_min(&chars, bstart, bend, &close_of, &mut replace, &mut remove) {
```
[the remark this answers is not in the document]
Comment (pmatos) [reply src/interpreter/builtins/regexp.rs]: Fixed in `7b09e21`.

Confirmed via Node + a parent-commit rebuild that this was a real regression (`/(a*|b)+/.exec("")` went from `["",""]` on the parent commit to `null` on this PR's original commit). You're right that the required first iteration of a `+` group is still allowed to match empty per spec 22.2.2.6.1 step 2.b — only later iterations are subject to the discard.

Scoped `nq_bump_bare_atom_min` to `*` groups only; `+` groups now fall through to the existing lazy-strip path, same as before this branch-aware rewrite existed. Also caught (via dual-evaluator review) that the same issue affected lazy nullable branches too (`/(a*?|b)+/.exec("")` had the same regression) — fixed by the same change. Added both as regression tests in `test262-extra/RegExp-nullable-alternation-quantifier.js`.

Validation: full test262 100.00% (99,568/99,568, 0 regressions), 341 unit tests, lint clean, test262-extra passing.

Note: `+`-repeated nullable-alternation branches that could otherwise *prefer* a consuming sibling (e.g. `/(a*|dc??)+/.exec("dc")`, expected `"d"`) remain unfixed — confirmed this was already broken on the parent commit too (pre-existing, not a regression), tracked in #373 alongside the other deferred nullable-alternation shapes.

_This assessment was made by two independent AI reviewers (Claude Sonnet 5 and Codex GPT-5.6). If you disagree, please reply and we'll re-evaluate._
